### ISSUE TITLE
Problems converting InstanceNormlization ONNX layer in TensorRT 7.1.3

### ISSUE THREAD

**Issue by FamousDirector** [NVIDIA/TensorRT#895]
## Description

Hello. I am trying to convert an ONNX model to TensorRT. It seems like converting to InstanceNormalization layer from ONNX is the problem.

## Environment

**TensorRT Version**: 7.1.3
**GPU Type**: Tesla T4
**Nvidia Driver Version**: 450.51.05
**CUDA Version**: 11.0
**CUDNN Version**: same as container
**Operating System + Version**: same as container
**Python Version (if applicable)**: N/A
**TensorFlow Version (if applicable)**: N/A
**PyTorch Version (if applicable)**: N/A
**Baremetal or Container (if container which image + tag)**: nvcr.io/nvidia/tensorrt:20.09-py3

## Relevant Files
Here is a [link](https://drive.google.com/file/d/1dcZ8Yo11jgPRRxv9_nImX_Xu3MJr0C6E/view) to the model I'm trying to convert.

## Steps To Reproduce
When using the following command:
`trtexec --onnx=osnet_ain_fused.onnx --minShapes=input:1x256x128x3 --optShapes=input:16x256x128x3 --maxShapes=input:32x256x128x3 --shapes=input:5x256x128x3 --saveEngine=osnet_ain_fused.engine`

I got the following error:
```
[11/11/2020-13:32:01] [I] [TRT] Detected 1 inputs and 1 output network tensors.
#assertion/home/jenkins/workspace/OSS/L0_MergeRequest/oss/plugin/instanceNormalizationPlugin/instanceNormalizationPlugin.cpp,305
Aborted (core dumped)
```
Some of the verbose output (before it crashes) is as follows:
```
[11/11/2020-13:29:26] [V] [TRT] *************** Autotuning format combination: Float(1,512) -> Float(1,512) ***************
[11/11/2020-13:29:26] [V] [TRT] --------------- Timing Runner: Relu_485 (Activation)
[11/11/2020-13:29:26] [V] [TRT] Tactic: 0 is the only option, timing skipped
[11/11/2020-13:29:26] [V] [TRT] Fastest Tactic: 0 Time: 0
[11/11/2020-13:29:26] [V] [TRT] Formats and tactics selection completed in 25.6369 seconds.
[11/11/2020-13:29:26] [V] [TRT] After reformat layers: 225 layers
[11/11/2020-13:29:26] [V] [TRT] Block size 268435456
[11/11/2020-13:29:26] [V] [TRT] Block size 16777216
[11/11/2020-13:29:26] [V] [TRT] Block size 16777216
[11/11/2020-13:29:26] [V] [TRT] Block size 16777216
[11/11/2020-13:29:26] [V] [TRT] Block size 4194304
[11/11/2020-13:29:26] [V] [TRT] Block size 4194304
[11/11/2020-13:29:26] [V] [TRT] Block size 4194304
[11/11/2020-13:29:26] [V] [TRT] Block size 4194304
[11/11/2020-13:29:26] [V] [TRT] Block size 4096
[11/11/2020-13:29:26] [V] [TRT] Block size 4096
[11/11/2020-13:29:26] [V] [TRT] Block size 4096
[11/11/2020-13:29:26] [V] [TRT] Block size 512
[11/11/2020-13:29:26] [V] [TRT] Block size 512
[11/11/2020-13:29:26] [V] [TRT] Total Activation Memory: 335557632
[11/11/2020-13:29:26] [I] [TRT] Detected 1 inputs and 1 output network tensors.
[11/11/2020-13:29:26] [V] [TRT] Conv_7 (scudnn) Set Tactic Name: volta_scudnn_128x64_relu_xregs_large_nn_v1
#assertion/home/jenkins/workspace/OSS/L0_MergeRequest/oss/plugin/instanceNormalizationPlugin/instanceNormalizationPlugin.cpp,305
Aborted (core dumped)
```

**Comment by pranavm-nvidia** [comment]
There was an assertion left over from before InstanceNorm supported dynamic shapes. I think that's probably what you're seeing here. I think it should work for you if you remove [this check in the plugin](https://github.com/NVIDIA/TensorRT/blob/master/plugin/instanceNormalizationPlugin/instanceNormalizationPlugin.cpp#L300-L308).

**Comment by FamousDirector** [comment]
Removing that section of code from the `v20.09` branch indeed worked. Model conversion was successful. @pranavm-nvidia is there anything else I can do?

**Comment by pranavm-nvidia** [comment]
I've removed the assertion upstream, so it should be fixed in the next release.

**Comment by FamousDirector** [comment]
@pranavm-nvidia okay great thanks. I've encountered a weird issues that stems from this same model however. When feeding data with batch size >7 the model returns incorrect results. Is that expected?

**Comment by pranavm-nvidia** [comment]
@FamousDirector That is not expected. Are the results for all instances incorrect, or just the ones after 7?
And is the behavior different if you use fixed shapes (i.e. set min == opt == max)?

**Comment by FamousDirector** [comment]
All instances are incorrect. I will need check again with static batching but I believe it was correct with static.

**Comment by FamousDirector** [comment]
I used the following command to generate a fixed input size `trtexec --onnx=osnet_ain_fused.onnx --shapes=input:32x256x128x3 --saveEngine=osnet_ain_fused.engine --explicitBatch`. This fixed shape has incorrect output as well. @pranavm-nvidia what are your thoughts?

**Comment by pranavm-nvidia** [comment]
I can't seem to run the model at all:
```
[11/13/2020-07:04:08] [E] [TRT] (Unnamed Layer* 494) [Shuffle]: at most one dimension may be inferred
ERROR: onnx2trt_utils.cpp:1517 In function scaleHelper:
[8] Assertion failed: dims.nbDims == 4 || dims.nbDims == 5
```
Are you making any modifications to the model before running it?

**Comment by FamousDirector** [comment]
@pranavm-nvidia I think you are running into a new issue I encountered in 7.2.1. https://github.com/onnx/onnx-tensorrt/issues/566

It works in the 20.09 container if you build with the TensorRT 20.09 release branch (and remove the assert as discussed above. I have the changes here: https://github.com/FamousDirector/TensorRT/tree/newbranch. I used the `/opt/tensorrt/install_open_source.sh` script and just made sure it pointed to my repo before building. Then `trtexec` converted it without issue.

**Comment by pranavm-nvidia** [comment]
@FamousDirector After fixing the [parser bug you mentioned](https://github.com/onnx/onnx-tensorrt/issues/566) I'm getting correct results on TensorRT 7.2 with larger batch sizes (I tried up to 32).

**Comment by FamousDirector** [comment]
@pranavm-nvidia fantastic thank you. I will try and get the dependencies together and try it myself. Will everything be included in the 20.11 release?

**Comment by pranavm-nvidia** [comment]
@FamousDirector Yes, this should be included in the 20.11 release.

**Comment by mk-nvidia** [comment]
Closing, thanks to @pranavm-nvidia for fixing the bugs.
@FamousDirector please let us know if you encounter more issues.

**Comment by FamousDirector** [comment]
Agreed this is fixed. Just confirmed in the last day. As a note however, the fix is not a part of the 20.11 release.
